src/Installer/Microsoft.Dotnet.Installation/Internal/DotnetArchiveDownloader.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.Dotnet.Installation;
+
+namespace Microsoft.Dotnet.Installation.Internal;
+
+/// <summary>
+/// Handles downloading and parsing .NET release manifests to find the correct installer/archive for a given installation.
+/// </summary>
+internal class DotnetArchiveDownloader(HttpClient httpClient) : IDisposable
```

Comment:
Do we need the constructor that specifies an HttpClient?  It doesn't look like we use that outside of the class, and there's no way to specify both an httpClient and a ReleaseManifest.

Reply:
we will when we want the library vs dnup to use a different custom agent header - or enable customization of timeout(s) for library callers